source/common/common/regex.cc

```diff
@@ -21,18 +21,18 @@ class CompiledStdMatcher : public CompiledMatcher {
 
   // CompiledMatcher
   bool match(absl::string_view value) const override {
-    try {
-      return std::regex_match(value.begin(), value.end(), regex_);
-    } catch (const std::regex_error& e) {
+    TRY_NEEDS_AUDIT { return std::regex_match(value.begin(), value.end(), regex_); }
```

Comment:
FWIW I don't think this one is going to be fixable.

Reply:
Yes, we should define a RAW_TRY to differentiate from try to me removed.

Reply:
The problem here is that this is called in route matchers when configured without safe_regex; is that right?

Should we enumerate the paths why this needs to be allowed?

The problem with adding RAW_TRY is that it gives devs an easy way to use exceptions where they shouldn't. In some cases it's unavoidable like this one. In those cases I think the best way would be to list this as a special-case in check_format.py. That way you have to edit that file, where we can leave behind a call saying why this pattern should be avoided if possible. RAW_TRY sounds too innocent. I'd rather people try to avoid having a throw occur in worker threads. WDYT?

Reply:
Makes sense. I am trying to figure out how to white a single try statement instead fo the entire source file.

Reply:
The mechanism for that is to exclude lines that contain `// NOLINT(try-assert-main)` from the lint check. There are a few examples of that in check_format.py. 

My reluctance to suggest that is because I don't want it to be that easy to override this check. I want someone to have to open up check_format.py and edit an allow-list, with a comment right above it saying why this is dangerous.

If we only need to do this for regex.cc it might be OK to include the whole file. WDYT?

Reply:
Sure, I can start from allowing raw try in regex.cc.